go-controller/pkg/ovn/secondary_layer3_network_controller.go

```diff
@@ -420,8 +420,12 @@ func (oc *SecondaryLayer3NetworkController) newRetryFramework(
 // Start starts the secondary layer3 controller, handles all events and creates all needed logical entities
 func (oc *SecondaryLayer3NetworkController) Start(ctx context.Context) error {
 	klog.Infof("Start secondary %s network controller of network %s", oc.TopologyType(), oc.GetNetworkName())
-
-	if err := oc.Init(ctx); err != nil {
+	networkID, err := oc.getNetworkID()
+	if err != nil {
+		return fmt.Errorf("unable to set networkID on secondary L3 controller for network %s, err: %w", oc.GetNetworkName(), err)
+	}
+	oc.networkID = &networkID
```

Comment:
this is weird. There seems like there layers of issues here. You oc.getNetworkID(). That function does:

```
func (oc *BaseSecondaryNetworkController) getNetworkID() (int, error) {
	if oc.networkID == nil || *oc.networkID == util.InvalidNetworkID {
		oc.networkID = ptr.To(util.InvalidNetworkID)
		nodes, err := oc.watchFactory.GetNodes()
		if err != nil {
			return util.InvalidNetworkID, err
		}
		*oc.networkID, err = util.GetNetworkID(nodes, oc.NetInfo)
		if err != nil {
			return util.InvalidNetworkID, err
		}
	}
	return *oc.networkID, nil
}

```

This function sets the value of oc.networkID within it. So there is no need for you to set it again here. Additionally the method is confusing, because getNetworkID() you would expect to only return the ID of the value, and not set it for the controller in the getter method.

Reply:
al'right removed setting the value again.